src/lib.rs

```diff
@@ -14,6 +14,7 @@ pub mod network;
 pub mod repair;
 pub mod shredder;
 pub mod slice;
+pub mod slice_index;
```

Comment:
maybe this should just be a sub-module of `slice`?

Reply:
or we have a different module that contains `slice`, `slot`, and `slice_index`

Reply:
I am fine with them being in different files as well as different modules, just think they should not all be at the top level.